lib/src/api/engine/any/mod.rs

```diff
@@ -1,87 +1,131 @@
 //! Dynamic support for any engine
 //!
-//! # Examples
+//! SurrealDB supports various ways of storing and accessing your data. For storing data we support a number of
+//! key value stores. These are RocksDB, SpeeDB, TiKV, FoundationDB and an in-memory store. We call these
+//! local engines. RocksDB and SpeeDB are file-based, single node key value stores. TiKV and FoundationDB are
+//! are distributed stores that can scale horizontally across multiple nodes. The in-memory store does not persist
+//! your data, it only stores it in memory. All these can be embedded in your application, so you don't need to
+//! spin up a SurrealDB server first in order to use them. We also support spinning up a server externally and then
+//! access your database via WebSockets or HTTP. We call these remote engines.
 //!
-//! ```no_run
-//! use serde::{Serialize, Deserialize};
-//! use serde_json::json;
-//! use std::borrow::Cow;
-//! use surrealdb::sql;
-//! use surrealdb::engine::any::connect;
-//! use surrealdb::opt::auth::Root;
+//! The Rust SDK abstracts away the implementation details of the engines to make them work in a unified way.
+//! All these engines, whether they are local or remote, work exactly the same way using the same API. The only
+//! difference in the API is the endpoint you use to access the engine. Normally you provide the scheme of the engine
+//! you want to use as a type parameter to `Surreal::new`. This allows you detect, at compile, whether the engine
+//! you are trying to use is enabled. If not, your code won't compile. This is awesome but it strongly couples your
+//! application to the engine you are using. In order to change an engine you would need to update your code to
+//! the new scheme and endpoint you need to use and recompile it. This is where the any engine comes in.
```

Comment:
```suggestion
//! you are trying to use is enabled. If not, your code won't compile. This is awesome but it strongly couples your
//! application to the engine you are using. In order to change an engine you would need to update your code to
//! the new scheme and endpoint you need to use and recompile it. This is where the any engine comes in.
```

Can this be in a `Note:` As it will need to stand out since it reads like a warning while making decisions

Reply:
I'm not sure Obinna, the intention wasn't to make it come across as a warning. It was more to highlight the pros and cons of both approaches. While the typed scheme approach strongly couples your application, it's also probably the safer bet out of the box because if it compiles that means the engine you want to use is enabled. That means your application is more likely to just work once you deploy it. This is why it's the approach we typically use in documentation. With `any` it's possible that you might forget to enable the engine you need while building your application. The compiler will happily compile your code, only to get an error when you try to run it.

Reply:
Ah, I see.  @rushmorem this is great explanation!